### PR TITLE
Random mode

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
#88 
#### What does this implement/fix? Explain your changes.
- `random_hex_color_gen` function added
- `color`,`bgcolor` and `projection` parameters random mode added
- `filter_color` function modified
- `filter_projection` function modified

### CODE REVIEW

File: samila/functions.py [hunk 1]
```diff
@@ -38,6 +38,16 @@ def random_equation_gen():
     return result
 
 
+def random_hex_color_gen():
+    """
+    Generate random hex color code.
+
+    :return: color code as str
+    """
+    random_color = "#%06x" % random.randint(0, 0xFFFFFF)
```
Comment (sadrasabouri) [inline samila/functions.py:47]: `0` and `0xFFFFFF` are both inclusive?

File: test/nft_upload_test.py [hunk 2]
```diff
@@ -11,10 +11,7 @@
 >>> g.plot()
 >>> NFT_STORAGE_API_KEY = os.environ["NFT_STORAGE_API_KEY"]
 >>> g.generate()
->>> random_projection = random.choice(list(Projection))
->>> random_color = random.choice(VALID_COLORS)
->>> random_bgcolor = random.choice(VALID_COLORS)
->>> g.plot(projection=random_projection,color=random_color,bgcolor=random_bgcolor)
+>>> g.plot(projection=Projection.RANDOM,color="random",bgcolor="random")
```
Comment (sadrasabouri) [inline test/nft_upload_test.py:14]: I prefer `g.plot(projection=Projection.RANDOM, color="random", bgcolor="random")`. This way we could keep our test's code style consistent.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
I assume minor edits are needed.

File: test/nft_upload_test.py (hunk 2 above)
Comment (sepandhaghighi) [reply test/nft_upload_test.py]: fixed in https://github.com/sepandhaghighi/samila/pull/124/commits/43bae2e88f413befba40f81b4dcc892f1e7a6912

File: samila/functions.py (hunk 1 above)
Comment (sepandhaghighi) [reply samila/functions.py:47]: Yeah.
Take a look at this [link](https://docs.python.org/3.6/library/random.html#random.randint)

**Pull request merged**